cogs/misc.py

```diff
@@ -188,21 +189,30 @@ async def random(self, ctx: Context, level: str, count: int = 3):
         async with ctx.typing():
             if count > 4 or count < 1:
                 raise commands.BadArgument("Number of songs must be between 1 and 4.")
-
+            
+            # Check whether input is level or constant
             try:
-                query_level = float(level.replace("+", ".5"))
+                if level.find('+') != -1:
+                    min_level = float(level.replace("+", ".5"))
+                    max_level = float(level.replace("+", ".9"))
+                elif level.find('.') == -1:
+                    min_level = level + ".0"
+                    max_level = level + ".4"
+                else:
+                    min_level = float(level)
+                    max_level = float(level)
             except:
-                raise commands.BadArgument("Invalid level provided.")
+                raise commands.BadArgument("Please enter a valid level or chart constant.")
 
             async with self.bot.db.execute(
                 "SELECT songs.title, songs.genre, songs.artist, songs.jacket, charts.difficulty, level, const, is_const_unknown, sdvxin.id AS sdvxin_id "
                 "FROM chunirec_charts charts "
                 "LEFT JOIN chunirec_songs songs ON charts.song_id = songs.id "
                 "LEFT JOIN sdvxin ON charts.song_id = sdvxin.song_id AND charts.difficulty = sdvxin.difficulty "
-                "WHERE level = ? OR const = ? "
+                "WHERE const >= ? AND const <= ? "
                 "ORDER BY random() "
                 "LIMIT ?",
-                (query_level, query_level, count),
+                (min_level, max_level, count),
```

Comment:
Not all songs (mostly new songs) have chart constants associated. This wouldn't work for such cases.

Perhaps this would work?
```py
try:
    if "." in level:
        query_level = float(level)
        where_clause = "WHERE const = ? "
    else:
        query_level = float(level.replace("+", ".5"))
        where_clause = "WHERE level = ? "
except ValueError:
    raise commands.BadArgument("Please enter a valid level or chart constant.")

async with self.bot.db.execute(
    "SELECT songs.title, songs.genre, songs.artist, songs.jacket, charts.difficulty, level, const, is_const_unknown, sdvxin.id AS sdvxin_id "
    "FROM chunirec_charts charts "
    "LEFT JOIN chunirec_songs songs ON charts.song_id = songs.id "
    "LEFT JOIN sdvxin ON charts.song_id = sdvxin.song_id AND charts.difficulty = sdvxin.difficulty "
    f"{where_clause}"
    "ORDER BY random() "
    "LIMIT ?",
    (query_level, count),
) as cursor:
     # snip
```